x-pack/legacy/plugins/apm/cypress/package.json

```diff
@@ -15,7 +15,7 @@
     "js-yaml": "^3.13.1",
     "p-limit": "^2.2.1",
     "ts-loader": "^6.1.0",
-    "typescript": "3.7.2",
+    "typescript": "3.8.1-rc",
```

Comment:
@elastic/apm-ui I need your help here to fix `APMRequestHandlerContext` which type is inferred as `any` 
![2020-02-17_10-40-24](https://user-images.githubusercontent.com/3198181/74642633-a2a48000-5173-11ea-8a02-ab4b49cb96ea.png)

Reply:
@dgieselaar fyi

Reply:
Looking into it!

Reply:
@restrry I can't explain this - there's no reason I can see why this fails, and others don't. It feels like a TS bug, but I cannot pinpoint it.

Reply:
btw - in my case, I don't see `context` as any, but the type for `context.params` is incorrect.

Reply:
I also don't see `context` as `any`
![image](https://user-images.githubusercontent.com/209966/74743132-2d62a900-5260-11ea-9f23-6c30c4188d1a.png)

Reply:
@sqren is `context.params` type correct in your case?

Reply:
Hmm. It seems like it's not. On master I get this:

![image](https://user-images.githubusercontent.com/209966/74749036-cea22d00-5269-11ea-864c-3f1075d279e2.png)

Reply:
I think it might be related to the fact that `Params` interface contains optional props only. 
When I remove an empty object default from https://github.com/elastic/kibana/blob/96a0aa0266dfd1a8b83b15dd8ae0efd273520cf1/x-pack/plugins/apm/server/routes/create_route.ts#L12
TS starts inferring context type, but not params
![2020-02-20_11-38-44](https://user-images.githubusercontent.com/3198181/74926693-979d5a00-53d6-11ea-875c-be1ffe99f131.png)

Reply:
I'm not here until Monday, and my only guess right now is that this is a TS bug (either in resolving or in showing an appropriate error). Can someone flag this with the TS team? If there's not much of a rush I can look into it Monday.

Reply:
@dgieselaar take your time, we are blocked by external dependencies as well